src/components/shared/TreeView.js

```diff
@@ -45,30 +49,34 @@ const TreeViewHeader = ({ fixedColumns, mainColumn }: TreeViewHeaderProps) =>
 
 function reactStringWithHighlightedSubstrings(
   string: string,
-  substring: string | null,
+  substrings: ?(string[]),
   className: string
 ) {
-  if (!substring) {
+  if (!substrings || !substrings.length) {
     return string;
   }
-  const lowercaseString = string.toLowerCase();
-  const result = [];
-  let startAt = 0;
-  let nextOccurrence = -1;
-  while (
-    (nextOccurrence = lowercaseString.indexOf(substring, startAt)) !== -1
-  ) {
-    const afterNextOccurrence = nextOccurrence + substring.length;
-    result.push(string.substring(startAt, nextOccurrence));
-    result.push(
-      <span key={nextOccurrence} className={className}>
-        {string.substring(nextOccurrence, afterNextOccurrence)}
+
+  const regexpStr = substrings.map(escapeStringRegexp).join('|');
+  const re = new RegExp(regexpStr, 'gi');
```

Comment:
mmm this is called at enough places to be relevant.
At the very least I can make sure we create it only once per render by passing a regexp instead of a string list to the TreeView component.

Then if we want to memoize it in the future we could just create a selector for it.